desktop_version/src/Music.cpp

```diff
@@ -92,81 +131,263 @@ class SoundTrack
     }
 
     void Dispose()
     {
-        /* Free stb_vorbis */
-        Mix_FreeMusic(m_music);
+        stb_vorbis_close(vorbis);
+        SDL_free(read_buf);
+        SDL_free(decoded_buf_playing);
+        SDL_free(decoded_buf_reserve);
+        if (!IsHalted()) {
+            FAudioVoice_DestroyVoice(musicVoice);
+            musicVoice = NULL;
+        }
     }
 
     bool Play(bool loop)
     {
         /* Create/Validate static FAudioSourceVoice, begin streaming */
-        if (Mix_PlayMusic(m_music, loop ? -1 : 0) == -1)
-        {
-            vlog_error("Mix_PlayMusic: %s", Mix_GetError());
+        shouldloop = loop;
+        sample_pos = 0;
+        stb_vorbis_seek_start(vorbis);
+        
+        if (IsHalted()) {
+            SDL_memset(&callbacks, '\0', sizeof(FAudioVoiceCallback));
+            callbacks.OnBufferStart = &MusicTrack::refillReserve;
+            callbacks.OnBufferEnd = &MusicTrack::swapBuffers;
+            FAudio_CreateSourceVoice(faudioctx, &musicVoice, &format, 0, 2.0f, &callbacks, NULL, NULL);
+        } else {
+            Pause();
+        }
+        FAudioSourceVoice_FlushSourceBuffers(musicVoice);
+
+        FAudioBuffer faudio_buffer;
+        SDL_memset(&faudio_buffer, 0, sizeof(FAudioBuffer));
+        if (looplength == 0) {
+            faudio_buffer.PlayLength = stb_vorbis_get_samples_float_interleaved(vorbis, channels, (float*)decoded_buf_playing, size/sizeof(float));
+        } else {
+            faudio_buffer.PlayLength = std::min(stb_vorbis_get_samples_float_interleaved(vorbis, channels, (float*)decoded_buf_playing, size/sizeof(float)), (loopbegin+looplength)-sample_pos);
+        }
+        faudio_buffer.AudioBytes = size;
+        faudio_buffer.pAudioData = decoded_buf_playing;
+        faudio_buffer.pContext = this;
+        sample_pos += faudio_buffer.PlayLength;
+        if (FAudioSourceVoice_SubmitSourceBuffer(musicVoice, &faudio_buffer, NULL)) {
+            vlog_error("Unable to queue sound buffer");
             return false;
         }
+        Resume();
         return true;
     }
 
     static void Halt()
     {
         /* FAudioVoice_Destroy */
-        Mix_HaltMusic();
+        if (!IsHalted()) {
+            FAudioSourceVoice_FlushSourceBuffers(musicVoice);
+            FAudioVoice_DestroyVoice(musicVoice);
+            musicVoice = NULL;
+        }
     }
 
     static bool IsHalted()
     {
-        /* return musicVoice == NULL; */
-        return Mix_PausedMusic() == 1;
+        return musicVoice == NULL;
     }
 
     static void Pause()
     {
-        /* FAudioSourceVoice_Pause */
-        Mix_PauseMusic();
+        vlog_info("paused musictracks");
+        if (!IsHalted()) FAudioSourceVoice_Stop(musicVoice, 0, FAUDIO_COMMIT_NOW);
     }
 
     static void Resume()
     {
-        /* FAudioSourceVoice_Resume */
-        Mix_ResumeMusic();
+        vlog_info("resumed musictracks");
```

Comment:
Do we *need* to log this? This looks like a temporary debug statement that was accidentally left in.